src/compas_model/algorithms/collider.py

```diff
@@ -0,0 +1,357 @@
+            + abs((box1.frame.zaxis * box1.height * 0.5).dot(axis))
+        )
+
+    # compute the obb collision
+    RPos = box1.frame.point - box0.frame.point
+
+    result = not (
+        GetSeparatingPlane(RPos, box0.frame.xaxis, box0, box1)
+        or GetSeparatingPlane(RPos, box0.frame.yaxis, box0, box1)
+        or GetSeparatingPlane(RPos, box0.frame.zaxis, box0, box1)
+        or GetSeparatingPlane(RPos, box1.frame.xaxis, box0, box1)
+        or GetSeparatingPlane(RPos, box1.frame.yaxis, box0, box1)
+        or GetSeparatingPlane(RPos, box1.frame.zaxis, box0, box1)
+        or GetSeparatingPlane(
+            RPos, box0.frame.xaxis.cross(box1.frame.xaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.xaxis.cross(box1.frame.yaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.xaxis.cross(box1.frame.zaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.yaxis.cross(box1.frame.xaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.yaxis.cross(box1.frame.yaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.yaxis.cross(box1.frame.zaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.zaxis.cross(box1.frame.xaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.zaxis.cross(box1.frame.yaxis), box0, box1
+        )
+        or GetSeparatingPlane(
+            RPos, box0.frame.zaxis.cross(box1.frame.zaxis), box0, box1
+        )
+    )
+
+    return result
+
+
+def face_to_face(polygons0, polygons1, tmax=1e-2, amin=1e1):
+    """Construct interfaces by intersecting coplanar mesh faces.
+
+    Parameters
+    ----------
+    assembly : compas_assembly.datastructures.Assembly
+        An assembly of discrete blocks.
+    nmax : int, optional
+        Maximum number of neighbours per block.
+    tmax : float, optional
+        Maximum deviation from the perfectly flat interface plane.
+    amin : float, optional
+        Minimum area of a "face-face" interface.
+
+    Returns
+    -------
+    Polygon of the Interface - :class:`compas.geometry.Polygon`
+    Current Element ID - list[int]
+    Other Element ID - list[int]
+    Current Element Face Index - int
+    Other Element Face Index - int
+    """
+
+    # --------------------------------------------------------------------------
+    # sanity check
+    # --------------------------------------------------------------------------
+    if shapely_available is False:
+        return []
+
+    # --------------------------------------------------------------------------
+    # iterate face polygons and get intersection area
+    # DEPENDENCY: shapely library
+    # --------------------------------------------------------------------------
+
+    def to_shapely_polygon(matrix, polygon, tmax=1e-3, amin=1e-1):
```

Comment:
why are all these defined inside the scope of `face_to_face`

Reply:
change to private methods and placed outside